server/src/controllers/job_applications/addJobApplication.js

```diff
@@ -0,0 +1,199 @@
+    // ! Query through the database with jobPostingId to see if it exists.
+
+    const jobPosting = await JobPosting.findByPk(parseInt(jobPostingId, 10));
+
+    // Check if the job posting was found
+    if (!jobPosting) {
+      // Return a 404 Not Found status if the job posting does not exist
+      return res.status(404).json({ error: "Job posting not found." });
+    }
+
+    // ! 400 status code with an error message is returned if the value is not one of the enums for application_status
+
+    const validApplicationStatuses = [
+      "Contacted",
+      "Rejected",
+      "R and I",
+      "Approved",
+      "In Progress",
+      "New",
+    ];
+
+    if (!validApplicationStatuses.includes(applicationStatus)) {
+      return res
+        .status(400)
+        .json({ error: "Invalid application status provided." });
+    }
+
+    // ! check statusInCanada and see if it fits the following enums: ENUM type (Citizen, PR, refugee, student visa, open work, other)
+
+    const validStatusInCanada = [
+      "Citizen",
+      "PR",
+      "Refugee",
+      "Student Visa",
+      "Open Work",
+      "Other",
+    ];
+
+    if (!validStatusInCanada.includes(statusInCanada)) {
+      return res
+        .status(400)
+        .json({ error: "Invalid status in Canada provided." });
+    }
+
+    // ! check that customResponses is an object.
+    // ! If it is not an object, return a 400 status code with an error message.
+
+    if (typeof customResponses !== "object") {
+      return res
+        .status(400)
+        .json({ error: "Custom responses must be an object." });
+    }
+
+    // ! Confirm that the job posting is still open before creating the job application
+    // ! Check that the date of the job application is before the closing date in the job posting model.
+
+    if (jobPosting.close_date < new Date()) {
+      return res.status(400).json({ error: "Job posting is closed." });
+    }
+
+    // ! Confirm that the postalCode is valid
+
+    if (!/^[A-Za-z]\d[A-Za-z][ -]?\d[A-Za-z]\d$/.test(postalCode)) {
+      return res.status(400).json({ error: "Invalid postal code." });
+    }
+
+    // ! Confirm the phone number is valid.
+
+    if (!/^\d{10}$/.test(phone)) {
+      return res.status(400).json({ error: "Invalid phone number." });
+    }
+
+    // ! Confirm that the email is valid.
+
+    if (!/^.+@.+\..+$/.test(email)) {
+      return res.status(400).json({ error: "Invalid email." });
+    }
+
+    // create the job application, and return the created job application object
+    // use the id of the job application to set the resume field as `${id}_${name}`
+
+    // ! Create the job application in the database
+
+    const jobApplication = await JobApplication.create({
+      job_posting_id: jobPostingId,
+      name,
+      email,
+      phone,
+      postal_code: postalCode,
+      resume: `temp`,
+      status_in_canada: statusInCanada,
+      application_status: applicationStatus,
```

Comment:
There is also a status_other field that needs to be included: https://github.com/uoftblueprint/tces/blob/main/server/src/models/job_applications.model.js#L61

Reply:
Hi! I think I just added the check for this in the most recent commit (e12a807)

Hope that helps!